src/v/net/server.h

```diff
@@ -146,6 +146,15 @@ class server {
     ss::abort_source& abort_source() { return _as; }
     bool abort_requested() const { return _as.abort_requested(); }
 
+    /**
+       Called by print_exceptional_future.
+       When this function returns 'true', the exception will be logged at WARN
+       level rather than the usual ERROR. This provides a customization point
+       for derived classes to avoid producing error logs for certain exception
+       types which may be application specific.
+     */
+    virtual bool is_message_error(std::exception_ptr) const { return false; };
```

Comment:
`is_auth_error` is called only once (in the above code).
`is_message_error` is also only called once.

Both of these have a rethrow in them.
I wonder if it would make sense to merge these two in a single function that returns the type of the exception, inline both of them or something along these lines just to not have multiple rethrows involved.

Reply:
removed